File: RailgunVB/Commands/JoinLeave/JoinLeaveRemove.vb
Imports Discord
Imports Discord.Commands
Imports RailgunVB.Core
Imports TreeDiagram
Imports TreeDiagram.Enums
Imports TreeDiagram.Models.Server

Namespace Commands.JoinLeave
    
    Partial Public Class JoinLeave
    
        <Group("remove")>
        Public Class JoinLeaveRemove
            Inherits SystemBase
            
            <Command("joinmsg")>
            Public Async Function JoinAsync(msg As Integer) As Task
                Await MsgHandlerAsync(msg, MsgType.Join)
            End Function
            
            <Command("leavemsg")>
            Public Async Function LeaveAsync(msg As Integer) As Task
                Await MsgHandlerAsync(msg, MsgType.Leave)
            End Function
            
            Private Async Function MsgHandlerAsync(index As Integer, type As MsgType) As Task
                If index < 0
                    await ReplyAsync("The specified Id can not be lower than 0.")
                    Return
                End If
                
                Dim data As ServerJoinLeave = Await Context.Database.ServerJoinLeaves.GetAsync(Context.Guild.Id)
                
                If data Is Nothing
                    Await ReplyAsync("Join/Leave has yet to be configured.")
                    Return
                ElseIf (type = MsgType.Join AndAlso data.JoinMessages.Count <= index) OrElse 
                       (type = MsgType.Leave AndAlso data.LeaveMessages.Count <= index)
                    Await ReplyAsync("Specified message is not listed.")
                    Return
                End If
                
                data.RemoveMessage(index, type)
                
                Await ReplyAsync($"Successfully removed from {Format.Bold(type.ToString())} messages.")
            End Function
            
        End Class
        
    End Class
    
End NameSpace

File: Railgun/Commands/JoinLeave/JoinLeaveRemove.cs
using System.Threading.Tasks;
using Discord;
using Finite.Commands;
using Railgun.Core;
using TreeDiagram;
using TreeDiagram.Enums;
using TreeDiagram.Models;
using TreeDiagram.Models.Server;

namespace Railgun.Commands.JoinLeave
{
	public partial class JoinLeave : SystemBase
	{
		[Alias("remove", "-")]
		public class JoinLeaveRemove : SystemBase
		{
			private ServerJoinLeave GetData(ulong guildId, bool create = false)
			{
				ServerProfile data;

				if (create)
					data = Context.Database.ServerProfiles.GetOrCreateData(guildId);
				else {
					data = Context.Database.ServerProfiles.GetData(guildId);

					if (data == null) 
						return null;
				}

				if (data.JoinLeave == null)
					if (create)
						data.JoinLeave = new ServerJoinLeave();
				
				return data.JoinLeave;
			}
			
			private Task MsgHandlerAsync(int index, MsgType type)
			{
				if (index < 0) return ReplyAsync("The specified Id can not be lower than 0.");

				var data = GetData(Context.Guild.Id);

				if (data == null) return ReplyAsync("Join/Leave has yet to be configured.");
				if ((type == MsgType.Join && data.JoinMessages.Count <= index) || (type == MsgType.Leave && data.LeaveMessages.Count <= index))
					return ReplyAsync("Specified message is not listed.");

				data.RemoveMessage(index, type);
				return ReplyAsync($"Successfully removed from {Format.Bold(type.ToString())} messages.");
			}

			[Command("joinmsg")]
			public Task JoinAsync(int msg) => MsgHandlerAsync(msg, MsgType.Join);

			[Command("leavemsg")]
			public Task LeaveAsync(int msg) => MsgHandlerAsync(msg, MsgType.Leave);
		}
	}
}